core/frontend/src/main/scala/io/udash/properties/seq/ForwarderSeqProperty.scala

```diff
@@ -1,19 +1,132 @@
+      originStructureListenerRegistration.cancel()
+      originStructureListenerRegistration = null
+    }
+  }
+
+  override def listenStructure(structureListener: (Patch[ElemType]) => Any): Registration = {
+    initOriginListeners()
+    structureListeners += structureListener
+    wrapListenerRegistration(new JsArrayRegistration(structureListeners, structureListener))
+  }
+
+  override def listen(valueListener: (Seq[B]) => Any, initUpdate: Boolean = false): Registration = {
+    initOriginListeners()
+    wrapListenerRegistration(super.listen(valueListener, initUpdate))
+  }
+
+  override def listenOnce(valueListener: (Seq[B]) => Any): Registration = {
+    initOriginListeners()
+    wrapListenerRegistration(super.listenOnce(valueListener))
+  }
+
+  protected def wrapListenerRegistration(reg: Registration): Registration = new Registration {
+    override def restart(): Unit = {
+      initOriginListeners()
+      reg.restart()
+    }
+
+    override def cancel(): Unit = {
+      reg.cancel()
+      killOriginListeners()
+    }
+
+    override def isActive(): Boolean =
+      reg.isActive()
+  }
 }
 
-trait ForwarderSeqProperty[A, +ElemType <: Property[A]]
-  extends ForwarderReadableSeqProperty[A, ElemType] with ForwarderProperty[Seq[A]] with SeqProperty[A, ElemType] {
+trait ForwarderWithLocalCopy[A, B, ElemType <: ReadableProperty[B], OrigType <: ReadableProperty[A]]
+  extends ForwarderReadableSeqProperty[A, B, ElemType, OrigType] {
+
+  import js.JSConverters._
+
+  protected var transformedElements: js.Array[ElemType] = js.Array()
+
+  protected def loadFromOrigin(): Seq[B]
+  protected def elementsFromOrigin(): Seq[ElemType]
+  protected def transformPatchAndUpdateElements(patch: Patch[OrigType]): Patch[ElemType]
+
+  override def get: Seq[B] = {
+    if (originListenerRegistration == null || !originListenerRegistration.isActive()) loadFromOrigin()
+    else transformedElements.map(_.get).toSeq
+  }
+
+  override def elemProperties: Seq[ElemType] = {
+    if (originListenerRegistration == null || !originListenerRegistration.isActive()) elementsFromOrigin()
+    else transformedElements.toSeq
+  }
+
+  override protected def onListenerInit(): Unit = {
+    val fromOrigin = elementsFromOrigin().toJSArray
+    if (transformedElements.map(_.id) != fromOrigin.map(_.id)) {
+      fireElementsListeners[ElemType](Patch[ElemType](0, transformedElements, fromOrigin, fromOrigin.isEmpty), structureListeners)
+      fireValueListeners()
+    } else if (transformedElements.map(_.get) != fromOrigin.map(_.get)) {
+      fireValueListeners()
+    }
+    transformedElements = fromOrigin
+  }
+
+  override protected def originListener(originValue: Seq[A]) : Unit = {
+    fireValueListeners()
+  }
+
+  override protected def originStructureListener(patch: Patch[OrigType]) : Unit = {
+    val transPatch = transformPatchAndUpdateElements(patch)
+    val cpy = structureListeners.jsSlice()
```

Comment:
This is not a copy, array views are shallow.

Reply:
From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/: 
If a new element is added to either array, the other array is not affected.

That's exactly what I want. I don't need to copy elements contained in array.

Reply:
Just needed that confirmation :ok_hand: